### PR TITLE
Refactor block syncing

### PR DESCRIPTION
**Pull request by ashchan**
Hopefully the refactored module will be simpler and robust.

### CODE REVIEW

**Comment by ashchan** [comment]
Notifications are handled by `NetworksController` now, which means `NetworksService` doesn't send subscription events anymore. All app networks change logic are done from `NetworksController`, and when it does that, it calls block sync renderer explicitly to stop/start syncing with the new network/chain info.

File: packages/neuron-wallet/src/controllers/networks/chain-info.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import initConnection from 'database/chain/ormconfig'
+import { updateMetaInfo, getMetaInfo } from 'database/chain/meta-info'
+import LockUtils from 'models/lock-utils'
+import logger from 'utils/logger'
+import DaoUtils from 'models/dao-utils'
+import { NetworkWithID, EMPTY_GENESIS_HASH } from 'types/network'
+import GetBlocks from 'block-sync-renderer/sync/get-blocks'
+
+// Open connection to a network and maintain chain info in database.
+export default class ChainInfo {
```
[the remark this answers is not in the document]
Comment (ashchan) [reply packages/neuron-wallet/src/controllers/networks/chain-info.ts]: Service is a bad layer most of which IMO should be removed.

[the remark this answers is not in the document]
Comment (ashchan) [reply packages/neuron-wallet/src/controllers/networks/chain-info.ts]: Also nothing except the networks controller would access this info - at least for now.

[the remark this answers is not in the document]
Comment (classicalliu) [reply packages/neuron-wallet/src/controllers/networks/chain-info.ts]: OK

**Pull request merged**